### PR TITLE
DSERVE-71-add-expression-matrix

### PR DESCRIPTION
**Pull request by keenangraham**

### CODE REVIEW

File: genomic_data_service/rnaseq/client.py [hunk 1]
```diff
@@ -3,5 +3,9 @@
 
 def add_rna_client(app):
     host = app.config.get('RNA_GET_ES')
-    es = Elasticsearch(host)
+    es = Elasticsearch(
+        host,
+        timeout=200,
```
Comment (pedrohr) [inline genomic_data_service/rnaseq/client.py]: this is a pretty large timeout, any issues with ES?

File: genomic_data_service/rnaseq/matrix.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+from flask import Response
+
+from genomic_data_service.reports.csv import CSVGenerator
+from genomic_data_service.searches.requests import make_search_request
+from snosearch.parsers import QueryString
+
+
+def extract_value(expression, key, default=''):
+    value = expression
+    for field in key:
```
Comment (pedrohr) [inline genomic_data_service/rnaseq/matrix.py:12]: can `key` be an array, tuple, dict, etc?

**Review by pedrohr — APPROVED** [review]

File: genomic_data_service/rnaseq/client.py (hunk 1 above)
Comment (keenangraham) [reply genomic_data_service/rnaseq/client.py]: Yeah, can probably reduce it. The default of `10` probably too short for first calculation of some expensive aggregations (ES does some agg caching on subsequent requests).

File: genomic_data_service/rnaseq/matrix.py (hunk 2 above)
Comment (keenangraham) [reply genomic_data_service/rnaseq/matrix.py:12]: Using tuple of strings but I suppose it could be any iterable. I guess an argument for type annotation.

**Pull request merged**